kubernetes/crd/dataset.json

```diff
@@ -0,0 +1,35 @@
+{
```

Comment:
during Istio development, we used protobuf to define APIs. I initially thought it was clunky, and still believe it was extremely expensive, although the cost of updating the API prevented many an error.

I have heard great things about this as a replacement. https://buf.build/pricing No idea if its a pile or not.

Reply:
I could be convinced of the benefits of an abstraction layer here at a later date, but today I am comfortable writing raw JSON which is applied directly against a k8s cluster. I don't really care, other than that I've seen the problems with YAML first-hand and don't want to do it again.

Reply:
Both `JSON` and `YAML` are terrible for specifying interfaces. They each have a different set of problems, the set are problems are hard to reason about, and the problems are hard to debug.

`YAML` is far more readable, though.